units/crons.go

```diff
@@ -902,13 +907,46 @@ func PopulateHostSetupJobs(env evergreen.Environment) amboy.QueueOperation {
 			return errors.Wrap(err, "error fetching provisioning hosts")
 		}
 
-		ts := utility.RoundPartOfMinute(30).Format(TSFormat)
+		sort.Slice(hosts, func(i, j int) bool {
+			return hosts[i].StartTime.Before(hosts[j].StartTime)
+		})
+
+		attemptsUsed := 0
+		jobsSubmitted := 0
+		collisions := 0
 		catcher := grip.NewBasicCatcher()
 		for _, h := range hosts {
-			catcher.Add(queue.Put(ctx, NewHostSetupJob(env, h, ts)))
+			if !h.IsContainer() {
+				if time.Since(h.StartTime) < 40*time.Second {
+					continue
+				}
+
+				if jobsSubmitted >= hostInitSettings.ProvisioningThrottle {
+					continue
+				}
+			}
+
+			err := queue.Put(ctx, NewHostSetupJob(env, &h))
+			if amboy.IsDuplicateJobError(err) {
+				collisions++
+				continue
+			}
+			catcher.Add(err)
+
+			attemptsUsed += h.ProvisionAttempts
+			jobsSubmitted++
 		}
 
-		catcher.Add(queue.Put(ctx, NewCloudHostReadyJob(env, ts)))
+		grip.Info(message.Fields{
+			"provisioning_hosts": len(hosts),
+			"throttle":           hostInitSettings.ProvisioningThrottle,
+			"jobs_submitted":     jobsSubmitted,
+			"total_attempts":     attemptsUsed,
+			"duplicates_seen":    collisions,
+			"message":            "host provisioning setup",
+		})
+
+		catcher.Add(queue.Put(ctx, NewCloudHostReadyJob(env, utility.RoundPartOfMinute(30).Format(TSFormat))))
```

Comment:
Shouldn't this take `ts` as an argument still?

Reply:
doesn't it?